File: lab4/hibernate_lab4/src/main/java/org/example/Product.java
package org.example;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
public class Product {
    @Id
    @GeneratedValue(
            strategy = GenerationType.AUTO)
    private int dbID;
    private String productName;
    private int unitsOnStock;
    @ManyToMany(mappedBy = "Products")
    private Set<Invoice> Invoices = new HashSet<>();

    public Product(){
    }

    public Product(String productName, int unitsOnStock){
        this.productName = productName;
        this.unitsOnStock = unitsOnStock;
    }

    public Set<Invoice> getInvoices(){
        return this.Invoices;
    }

    public void addInvoice(Invoice invoice){
        this.Invoices.add(invoice);
    }

    public String toString(){
        return this.productName;
    }

}

File: lab3/KacperCwiertniaEFProducts/KacperCwiertniaEFProducts/Product.cs
using System;
namespace KacperCwiertniaEFProducts
{
	public class Product
	{
        public Product(){
            this.Invoices = new HashSet<Invoice>();
        }

        public int ProductID { get; set; }
		public string ProductName { get; set; }
		public int UnitsOnStock { get; set; }
        public virtual ICollection<Invoice> Invoices { get; set; }
    }
}

